spec/checkNewCronJobSpec.js

```diff
@@ -0,0 +1,444 @@
+    patch:
+     '@@ -0,0 +1,25 @@\n+class CronJobTests(test_utils.GenericTestBase):\n+' +
+      'FIVE_WEEKS = datetime.timedelta(weeks=5)\n+    NINE_WEEKS' +
+      '= datetime.timedelta(weeks=9)\n+\n+    def setUp(self):\n+' +
+      '       super(CronJobTests, self).setUp()\n+ ' +
+      ' self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)\n+' +
+      'self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)\n+' +
+      'self.set_admins([self.ADMIN_USERNAME])\n+' +
+      ' self.testapp_swap = self.swap(\n+' +
+      'self, \'testapp\', webtest.TestApp(main_cron.app))\n+\n+' +
+      'self.email_subjects = []\n+        self.email_bodies = []\n+' +
+      'def _mock_send_mail_to_admin(email_subject, email_body):\n+ ' +
+      'ocks email_manager.send_mail_to_admin() as it\'s not possible to\n+' +
+      'send mail with self.testapp_swap, i.e with the URLs defined in\n+' +
+      ' main_cron.\n+            """\n+            self.email_subjects' +
+      '.append(email_subject)\n+            self.email_bodies.append' +
+      '(email_body)\n+\n+        self.send_mail_to_admin_swap = self.swap' +
+      '(\n+            email_manager, \'send_mail_to_admin\',' +
+      '_mock_send_mail_to_admin)\n'
+  };
+
+  const nonJobFile = {
+    sha: 'd144f32b9812373d5f1bc9f94d9af795f09023ff',
+    filename: 'core/domain/exp_fetchers.py',
+    status: 'added',
+    additions: 1,
+    deletions: 0,
+    changes: 1,
+    blob_url:
+      'https://github.com/oppia/oppia/blob/67fb4a973b318882af3b5a894130' +
+      'e110d7e9833c/core/domain/exp_fetchers.py',
+    raw_url:
+      'https://github.com/oppia/oppia/raw/67fb4a973b318882af3b5a894130' +
+      'e110d7e9833c/core/domain/exp_fetchers.py',
+    contents_url:
+      'https://api.github.com/repos/oppia/oppia/contents/core/domain/' +
+      'exp_fetchers.py?ref=67fb4a973b318882af3b5a894130e110d7e9833c',
+    patch: '@@ -0,0 +1 @@\n+# def _migrate_states_schema' +
+      '(versioned_exploration_states, exploration_id):',
+  };
+
+  beforeEach(() => {
+    spyOn(scheduler, 'createScheduler').and.callFake(() => { });
+
+    github = {
+      issues: {
+        createComment: jasmine.createSpy('createComment').and.returnValue({}),
+        addLabels: jasmine.createSpy('addLabels').and.returnValue({}),
+        addAssignees: jasmine.createSpy('addAssignees').and.returnValue({})
+      },
+    };
+
+    robot = createProbot({
+      id: 1,
+      cert: 'test',
+      githubToken: 'test',
+    });
+
+    app = robot.load(oppiaBot);
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(checkCronJobModule, 'checkForNewCronJob').and.callThrough();
+    spyOn(checkPullRequestBranchModule, 'checkBranch').and.callFake(() => { });
+    spyOn(checkWIPModule, 'checkWIP').and.callFake(() => { });
+    spyOn(
+      checkPullRequestTemplateModule, 'checkTemplate'
+    ).and.callFake(() => { });
+    spyOn(newCodeOwnerModule, 'checkForNewCodeowner').and.callFake(() => { });
```

Comment:
You need to mock all the modules that listen on the open event. Eg ApiForSheets, pullRequestJobCheck, etc.

Reply:
done